File: sdk/src/main/java/org/p2p/solanaj/kits/TokenAccounts.java
package org.p2p.solanaj.kits;

import java.util.List;

import com.squareup.moshi.Json;

import org.p2p.solanaj.rpc.types.RpcResultObject;

public class TokenAccounts extends RpcResultObject {

    public static class Account {
        @Json(name = "account")
        private AccountInfoParsed account;
        @Json(name = "pubkey")
        private String pubkey;

        public AccountInfoParsed getAccount() {
            return account;
        }

        public String getPubkey() {
            return pubkey;
        }
    }

    @Json(name = "value")
    private List<Account> value;

    public List<Account> getAccounts() {
        return value;
    }
}

File: solana/src/main/java/org/p2p/solanaj/model/types/TokenAccounts.kt
package org.p2p.solanaj.model.types

import com.google.gson.annotations.SerializedName
import org.p2p.solanaj.kits.AccountInfoParsed

data class TokenAccounts(
    @SerializedName("value")
    val accounts: List<Account>
)

data class Account(
    @SerializedName("account")
    val account: AccountInfoParsed,

    @SerializedName("pubkey")
    val pubkey: String
) {
    val tokenMintAddress: String
        get() = account.data.parsed.info.mint
}
